src/platform/qt/library/LibraryGrid.cpp

```diff
@@ -0,0 +1,93 @@
+/* Copyright (c) 2014-2017 waddlesplash
+ *
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+#include "LibraryGrid.h"
+
+#include "ui_LibraryGrid.h"
+
+namespace QGBA {
+
+// Game banner image size
+static const quint32 GRID_BANNER_WIDTH = 144;
+static const quint32 GRID_BANNER_HEIGHT = 48;
+
+static const quint32 ICON_BANNER_WIDTH = 64;
+static const quint32 ICON_BANNER_HEIGHT = 64;
+
+LibraryGrid::LibraryGrid(QWidget* parent_widget)
+	: QListWidget(parent_widget)
+	, m_ui(new Ui::LibraryGrid)
+{
+	m_ui->setupUi(this);
+	setViewStyle(LibraryStyle::STYLE_GRID);
+
+	connect(this, &QListWidget::itemActivated, this, &LibraryGrid::StartGame);
+}
+LibraryGrid::~LibraryGrid() {
+	for (QListWidgetItem* i : m_items.keys()) { // TODO: probably not needed
+		delete i;
+	}
+	delete m_ui;
```

Comment:
Is there no way to automatically parent m_ui to this object so it gets automatically deleted? unique_ptr maybe?